pkg/sdkserver/sdkserver.go

```diff
@@ -330,6 +331,46 @@ func (s *SDKServer) Ready(ctx context.Context, e *sdk.Empty) (*sdk.Empty, error)
 	return e, nil
 }
 
+// Allocate set the GameServer to Allocate, as longs as it's not in UnHealthy,
+// Shutdown or has a DeletionTimeStamp(). Times out after 30 seconds if it cannot
+// complete the operation due to contention issues.
+func (s *SDKServer) Allocate(context.Context, *sdk.Empty) (*sdk.Empty, error) {
+	s.logger.Info("Received self Allocate request")
+	err := wait.PollImmediate(500*time.Millisecond, 30*time.Second, func() (done bool, err error) {
```

Comment:
wait.ExponentialBackoff() is more appropriate to handle contention issues

having said that, I don't see why this would ever be contending - the GS is running so unlikely to be Unhealthy and not in Shutdown and not deleted

Reply:
YES! You are right. 

I was worried about the Exponential backoff potentially going for a really long time, but I can provide a [Backoff](https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Backoff) with a `Cap`, and I get the same thing. I will make this change. Nice catch - I was looking for this, but didn't find it previously :+1: